Marketplace/photoFrame/digitalFrame.js

```diff
@@ -0,0 +1,67 @@
+//
+// digitalFrame
+//
+//  A digital picture frame 
+//  Created by Liv Erickson on 10/11/2018
+//  Copyright 2018 High Fidelity, Inc.
+//
+// Distributed under the Apache License, Version 2.0.
+// See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+(function(){
+
+    // Set up the default transition speed and photo array
+    var DEFAULT_TRANSITION_SPEED = 10; // seconds
+    var DEFAULT_PHOTO_ARRAY = [
+        Script.resolvePath('images/image1.jpg'),
+        Script.resolvePath('images/image2.jpg'),
+        Script.resolvePath('images/image3.jpg'),
+        Script.resolvePath('images/image4.jpg'),
+        Script.resolvePath('images/image5.jpg')
+    ];
+
+    // We will convert the seconds to milliseconds for our Script.interval call
+    var SECONDS_TO_MILLISECONDS = 1000; 
+
+    // Variables in place to store user-specified data
+    var photos = [];
+    var activeIndex = 0;
+    var speed;
+
+    // Keep a reference to the Image Entity entityID for editing
+    var _entityID;
+
+    // Change our photos based on the photos array and change speed
+    // Return to the start after going through all our photos
+    var changePhotos = function() {
+        if (activeIndex < photos.length) {
+            Entities.editEntity(_entityID, {'textures' : JSON.stringify({'tex.picture' : photos[activeIndex]})});
+            activeIndex++;
+        } else {
+            Entities.editEntity(_entityID, {'textures' : JSON.stringify({'tex.picture' : photos[0]})});
+            activeIndex = 1;
+        }
+    };
+
+    // The preload function is called when the script is added, reloaded, or the server starts up
+    this.preload = function(entityID) {
+        _entityID = entityID;
+        try {
+            var customProperties = JSON.parse(Entities.getEntityProperties(_entityID, 'userData').userData).digitalFrame;
+            photos = customProperties.photos;
+            speed = customProperties.changeSpeed * SECONDS_TO_MILLISECONDS;
+            Script.setInterval(changePhotos, speed);
```

Comment:
I tested this implementation of Script.setInterval and Script.clearInterval without saving the returned reference to the QTimer. I was unable to clear the interval in my client. Does the server script behave differently? 
I suggest saving the QTimer reference with a global variable intervalID to later use in unload's Script.clearInterval function. 

// As a global variable 
var intervalID = null;

// Replace line with:
intervalID = Script.setInterval(changePhotos, speed);

Reply:
As written, this still stops when the script is removed, but does through the incompatible type error. Not sure why it works, but will make the suggested change anyway.